phonon/update.py

```diff
@@ -104,19 +106,18 @@ def end_session(self, block=True):
         possible.
         """
         if not self.ref.dereference(self.__execute, block=block):
-        	with self.ref.lock(block=block):
-	            if self.is_expired() or self.ref.force_expiry:
-	                # If this update has expired but other active references
-	                # to the resource still exist, we force this update to
-	                # execute. We reset the time_modified_key and cached data
-	                # to prevent any other processes from executing the same
-	                # data.
-	                self.__execute()
-	                self.__process.client.delete(self.ref.resource_key)
-	                self.__process.client.set(self.ref.times_modified_key, 0)
-	            else:
-	                self.__cache()
-
+            if self.is_expired():
+                # If this update has expired but other active references
+                # to the resource still exist, we force this update to
+                # execute. We reset the time_modified_key and cached data
+                # to prevent any other processes from executing the same
+                # data.
+                self.__execute()
+                with self.ref.lock(block=block):
+                    self._BaseUpdate__process.client.delete(self.ref.resource_key)
```

Comment:
This is only ok in tests... It's not guaranteed to be supported. Just import the client and create an instance local to the method, or scoped to the instance if performance is a concern.